### PR TITLE
upgrade a backported patch

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR tries to fix a problem with a privately backported feature. The columns added for the feature are not added idem potent so people can not backport them. I propose that all DB alteration from here on in will be done with the `IDEM_POTENT_...()` set of stored procedures that we have to prevent these kind of issues for users.

So far this has only been done with one feature but I propose to regirously do this for all code.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]
LGTM (but hey this only fixes if someone upgrades to 4.17.1.0 or 4.18.0.0++, it may break still for 4.17.0.1, 4.17.0.0 releases)

File: engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql [hunk 1]
```diff
@@ -887,9 +887,8 @@ left join `cloud`.`mshost_status` on
     ((`cloud`.`mshost`.`uuid` = `cloud`.`mshost_status`.`ms_id`)));
 
 -- Alter event table to add resource_id and resource_type
-ALTER TABLE `cloud`.`event`
-    ADD COLUMN `resource_id` bigint unsigned COMMENT 'ID of the resource associated with the event' AFTER `domain_id`,
-    ADD COLUMN `resource_type` varchar(32) COMMENT 'Type of the resource associated with the event' AFTER `resource_id`;
+CALL `cloud`.`IDEMPOTENT_ADD_COLUMN`('cloud.event','resource_id', 'bigint unsigned COMMENT "ID of the resource associated with the event" AFTER `domain_id`');
```
Comment (rohityadavcloud) [inline engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql:890]: @DaanHoogland is this IDEMPOTENT_ADD_COLUMN  method declared in this sql ?

Comment (DaanHoogland) [reply engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql:890]: sure, but also it should already be present in the server. We have re-defined it in several upgrade steps.

**Review by borisstoyanov — APPROVED** [review]
LGTM, just upgraded this manually and it's working fine

**Pull request merged**